### ISSUE TITLE
`separation` of TelescopeFrame is broken

### ISSUE THREAD

**Issue by MaxNoe** [cta-observatory/ctapipe#900]
```
from ctapipe.coordinates import HorizonFrame, TelescopeFrame
from astropy.coordinates import SkyCoord
import astropy.units as u
 
p = SkyCoord(az=0 * u.deg, alt=1 * u.deg, frame=HorizonFrame())
t1 = SkyCoord(0 * u.deg, 1 * u.deg, frame=TelescopeFrame(pointing_direction=p))
t2 = SkyCoord(0 * u.deg, 0 * u.deg, frame=TelescopeFrame(pointing_direction=p))
print(t1.separation(t2))
print(t1.separation3d(t2))
```

The first is 90° and the second raises an error about not being able to convert degrees to meter.

**Comment by MaxNoe** [comment]
Might be solvable by #899

**Comment by MaxNoe** [comment]
Fixed in #896
